scylla/src/cluster/mod.rs

```diff
@@ -1,3 +1,20 @@
+//! This module holds entities that represent the cluster as a whole,
+//! nodes in the cluster (together with a pool of connections),
+//! the cluster's state, and logic for ruling out specific nodes.
+//!
+//! This includes:
+//! - node's representation (Node),
+//! - metadata representation, fetching and management, including:
+//!     - topology metadata,
+//!     - schema metadata,
+//!     - tablet metadata,
+//! - ClusterData, which is a snapshot of the cluster's state.
+//!   - ClusterData is replaced atomically upon a metadata refresh,
+//!     preventing any issues arising from mutability, including races.
+//! - HostFilter, which is a way to filter out some nodes and thus
+//!   not contact them at all on any condition.
+//!
+
```

Comment:
Commit: "cluster: add module docstring"

🔧 Again there are some internal details - please put them in a place that is not visible in a public documentation.

Reply:
As agreed in person, this particular docstring is OK.